specs/dotsigstore_bundle/verifier.go

```diff
@@ -0,0 +1,115 @@
+// Pseudo-code for an example verification flow for a sigstore bundle,
+// with some input parameters required from the client.
+
+type Verifier interface{
+	// Verify verifier the provided data against the signatre,
+	// optionally relying on the kid as hint to which key to use.
+	Verify(data, sig []byte, kid string) error
+}
+
+
+type Options {
+	// Verifier for user provided keys. May know about many keys.
+	Verifier Verifier
+	// Optional user provided Root CA
+	RootCA *x509.Certificate
+	// Optional Rekor URL
+	RekorURL string
+	// Optional Fulcio URL
+	FulcioURL string
+}
+
+// Verify is the main verirication function.
+// It accepts a required sigstore bundle, and an optional path to the blob.
+// In if the attestation in the bundle is a DSSE envelope, the blob path
+// is ignored.
+func Verify(b *bundle, blob string, opts Options) error {
+	if b.AttestationDsse != nil {
+		if err = verifyDsse(b, opts); err != nil {
+			return err
+		}
+	} else {
+		if err = verifyBlob(b, blob, opts); err != nil {
+			return err
+		}
+	}
+
+	if b.rekorEntry != nil {
+		err = verifyRekor(b, blob, opts); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// Extract the verifier to use, either based on a set of provided keys, or
+// a complete X509 chain.
+func getVerifier(b *bundle, opts Options) Verifier {
+	if b.publicKey {
+		return = opts.Verifier
+	} else {
+		return = x509Verifier(b.X509Cert, opts)
+	}
+}
+
+// Standard DSSE verification using the provided verification material.
+func verifyDsse(b *bundle, opts options) error {
+	verifier = getVerifier(b, opts)
+	return stdDsseVerification(b.AttestationDsse, verifier)
```

Comment:
Where's `stdDsseVerification` defined?

Reply:
I believe that this is intended just to be pseudo-code, so `stdDsseVerification` is a short-hand for "insert the [standard guidelines for DSSE verification](https://github.com/secure-systems-lab/dsse/blob/master/protocol.md#dsse-protocol) here"

Reply:
Yes, what @bdehamer said, pseudo-code, so some commonly expected functions are not listed, the same for the `loadCAFromTuf` etc in the comment below.

Reply:
Ah gotcha!